cmd/brew-test-bot.rb

```diff
@@ -977,7 +980,10 @@ def test_ci_upload(tap)
 
     if (pr = ENV["UPSTREAM_PULL_REQUEST"])
       pull_pr = "https://github.com/#{tap.user}/homebrew-#{tap.repo}/pull/#{pr}"
-      safe_system "brew", "pull", "--clean", pull_pr
+      args = []
+      args << "--tap=#{@tap}" unless @tap.nil?
+      args << pull_pr
+      test "brew", "pull", "--clean", *args
```

Comment:
> brew pull --tap=homebrew/science https://github.com/Linuxbrew/homebrew-science/pull/342

That that works is at best an implementation detail and at worst a bug that we'll fix at a later point so I wouldn't rely on that behaviour.

> The motivation is to build bottles using test-bot whose INSTALL_RECEIPT.json has "tap":"homebrew/science" rather than "tap":"linuxbrew/science"

In that case you should adjust your bottle-building tooling to change the `INSTALL_RECEIPT.json` to produce the output you wish. It does seem, though, that having a dedicated Linuxbrew/homebrew-science tap is a much more sensible long-term decision. Again, that Homebrew/homebrew-science supports Linux is currently a lucky accident in which I'm currently looking the other way but the prevalence of `if OS.mac?` in that tap is not a good, maintainable long-term solution.